### PR TITLE
port commands

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Ports the abandoned colony changes to 1.11
- Seems also to add some changes we have left behind before.

Review please

### CODE REVIEW

**Comment by marvin-bitterlich** [comment]
:shipit: this needs some formatting but is fine

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1101/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 1]
```diff
@@ -182,6 +199,11 @@
     private double overallHappiness = 5;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java:199]: MAJOR Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/colony/ColonyView.java [hunk 2]
```diff
@@ -78,6 +78,11 @@
     private double overallHappiness = 5;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/ColonyView.java:78]: MAJOR Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/commands/colonycommands/ChangeColonyOwnerCommand.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+    @NotNull
+    @Override
+    public String getCommandUsage(@NotNull final ICommandSender sender)
+    {
+        return super.getCommandUsage(sender) + "<ColonyId> <(Optional)Player>";
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/colonycommands/ChangeColonyOwnerCommand.java:52]: MAJOR Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

File: src/main/java/com/minecolonies/coremod/commands/generalcommands/RandomTeleportCommand.java [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+    /**
+     * Method used to teleport the player.
+     * @param sender           the sender to have access to the world.
+     * @param playerToTeleport the player which shall be teleported.
+     */
+    private static void teleportPlayer(final ICommandSender sender, final EntityPlayer playerToTeleport)
+    {
+        //Now the position will be calculated, we will try up to 4 times to find a save position.
+        int attCounter = 0;
+        while (attCounter <= ATTEMPTS)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/generalcommands/RandomTeleportCommand.java:132]: MAJOR Reduce the total number of break and continue statements in this loop to use at most one. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS135)

File: src/main/java/com/minecolonies/coremod/commands/citizencommands/AbstractCitizensCommands.java [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    @NotNull
+    @Override
+    public String getCommandUsage(@NotNull final ICommandSender sender)
+    {
+        return super.getCommandUsage(sender) + "<ColonyId> <CitizenId>";
+    }
+
+    @NotNull
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/citizencommands/AbstractCitizensCommands.java:45]: MAJOR Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

File: src/main/java/com/minecolonies/coremod/commands/colonycommands/ShowColonyInfoCommand.java [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+    @NotNull
+    @Override
+    public String getCommandUsage(@NotNull final ICommandSender sender)
+    {
+        return super.getCommandUsage(sender) + "<ColonyId|OwnerName>";
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/colonycommands/ShowColonyInfoCommand.java:57]: MAJOR Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 53 issues
* ![MAJOR][MAJOR] 25 major
* ![MINOR][MINOR] 27 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/bc451629887e96c55c649af68bf450e431fc1bf8/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.260 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L48](https://github.com/Minecolonies/minecolonies/blob/bc451629887e96c55c649af68bf450e431fc1bf8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L48): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L48](https://github.com/Minecolonies/minecolonies/blob/bc451629887e96c55c649af68bf450e431fc1bf8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L48): class "Colony" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L597](https://github.com/Minecolonies/minecolonies/blob/bc451629887e96c55c649af68bf450e431fc1bf8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L597): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1076](https://github.com/Minecolonies/minecolonies/blob/bc451629887e96c55c649af68bf450e431fc1bf8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1076): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1077](https://github.com/Minecolonies/minecolonies/blob/bc451629887e96c55c649af68bf450e431fc1bf8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1077): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1201](https://github.com/Minecolonies/minecolonies/blob/bc451629887e96c55c649af68bf450e431fc1bf8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1201): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1203](https://github.com/Minecolonies/minecolonies/blob/bc451629887e96c55c649af68bf450e431fc1bf8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1203): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1205](https://github.com/Minecolonies/minecolonies/blob/bc451629887e96c55c649af68bf450e431fc1bf8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1205): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1398](https://github.com/Minecolonies/minecolonies/blob/bc451629887e96c55c649af68bf450e431fc1bf8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1398): Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 62 issues
* ![MAJOR][MAJOR] 32 major
* ![MINOR][MINOR] 27 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/49140259e9454b4d2b76abbc90a68fa41457c9f6/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.260 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L48](https://github.com/Minecolonies/minecolonies/blob/49140259e9454b4d2b76abbc90a68fa41457c9f6/src/main/java/com/minecolonies/coremod/colony/Colony.java#L48): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L48](https://github.com/Minecolonies/minecolonies/blob/49140259e9454b4d2b76abbc90a68fa41457c9f6/src/main/java/com/minecolonies/coremod/colony/Colony.java#L48): class "Colony" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L597](https://github.com/Minecolonies/minecolonies/blob/49140259e9454b4d2b76abbc90a68fa41457c9f6/src/main/java/com/minecolonies/coremod/colony/Colony.java#L597): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1076](https://github.com/Minecolonies/minecolonies/blob/49140259e9454b4d2b76abbc90a68fa41457c9f6/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1076): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1077](https://github.com/Minecolonies/minecolonies/blob/49140259e9454b4d2b76abbc90a68fa41457c9f6/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1077): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1201](https://github.com/Minecolonies/minecolonies/blob/49140259e9454b4d2b76abbc90a68fa41457c9f6/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1201): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1203](https://github.com/Minecolonies/minecolonies/blob/49140259e9454b4d2b76abbc90a68fa41457c9f6/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1203): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1205](https://github.com/Minecolonies/minecolonies/blob/49140259e9454b4d2b76abbc90a68fa41457c9f6/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1205): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1398](https://github.com/Minecolonies/minecolonies/blob/49140259e9454b4d2b76abbc90a68fa41457c9f6/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1398): Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/commands/citizencommands/AbstractCitizensCommands.java [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+        }
+        return citizenId;
+    }
+
+    /**
+     * Citizen commands have to overwrite this to handle their specialized code.
+     * @param colonyId  the id for the colony
+     * @param citizenId  the id for the citizen
+     */
+    abstract void executeSpecializedCode(@NotNull final MinecraftServer server, final ICommandSender sender, final Colony colonyId, final int citizenId);
```
Comment (cltnschlosser) [inline src/main/java/com/minecolonies/coremod/commands/citizencommands/AbstractCitizensCommands.java:154]: Make this protected

File: src/main/java/com/minecolonies/coremod/commands/colonycommands/ListColoniesCommand.java [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+    @NotNull
+    @Override
+    public List<String> getTabCompletionOptions(
+            @NotNull final MinecraftServer server,
+            @NotNull final ICommandSender sender,
+            @NotNull final String[] args,
+            @Nullable final BlockPos pos)
+    {
+        return new ArrayList<>();
+    }
```
Comment (cltnschlosser) [inline src/main/java/com/minecolonies/coremod/commands/colonycommands/ListColoniesCommand.java:142]: Collection.emptyList()

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 62 issues
* ![MAJOR][MAJOR] 32 major
* ![MINOR][MINOR] 27 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/37873e7144fd9aa45ed0ba562f0f7a9ab2f2a014/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.260 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L48](https://github.com/Minecolonies/minecolonies/blob/37873e7144fd9aa45ed0ba562f0f7a9ab2f2a014/src/main/java/com/minecolonies/coremod/colony/Colony.java#L48): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L48](https://github.com/Minecolonies/minecolonies/blob/37873e7144fd9aa45ed0ba562f0f7a9ab2f2a014/src/main/java/com/minecolonies/coremod/colony/Colony.java#L48): class "Colony" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L597](https://github.com/Minecolonies/minecolonies/blob/37873e7144fd9aa45ed0ba562f0f7a9ab2f2a014/src/main/java/com/minecolonies/coremod/colony/Colony.java#L597): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1076](https://github.com/Minecolonies/minecolonies/blob/37873e7144fd9aa45ed0ba562f0f7a9ab2f2a014/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1076): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1077](https://github.com/Minecolonies/minecolonies/blob/37873e7144fd9aa45ed0ba562f0f7a9ab2f2a014/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1077): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1201](https://github.com/Minecolonies/minecolonies/blob/37873e7144fd9aa45ed0ba562f0f7a9ab2f2a014/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1201): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1203](https://github.com/Minecolonies/minecolonies/blob/37873e7144fd9aa45ed0ba562f0f7a9ab2f2a014/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1203): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1205](https://github.com/Minecolonies/minecolonies/blob/37873e7144fd9aa45ed0ba562f0f7a9ab2f2a014/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1205): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1398](https://github.com/Minecolonies/minecolonies/blob/37873e7144fd9aa45ed0ba562f0f7a9ab2f2a014/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1398): Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by xavierh** [comment]
LGTM 
What about doing the final ?

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1101/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by Raycoms** [comment]
Will run an automatic final assignment task on the project this week anyway.

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 62 issues
* ![MAJOR][MAJOR] 32 major
* ![MINOR][MINOR] 27 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/945b86d195fe355ee006c7c41de56eafe80af905/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.260 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L48](https://github.com/Minecolonies/minecolonies/blob/945b86d195fe355ee006c7c41de56eafe80af905/src/main/java/com/minecolonies/coremod/colony/Colony.java#L48): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L48](https://github.com/Minecolonies/minecolonies/blob/945b86d195fe355ee006c7c41de56eafe80af905/src/main/java/com/minecolonies/coremod/colony/Colony.java#L48): class "Colony" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L597](https://github.com/Minecolonies/minecolonies/blob/945b86d195fe355ee006c7c41de56eafe80af905/src/main/java/com/minecolonies/coremod/colony/Colony.java#L597): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1076](https://github.com/Minecolonies/minecolonies/blob/945b86d195fe355ee006c7c41de56eafe80af905/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1076): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1077](https://github.com/Minecolonies/minecolonies/blob/945b86d195fe355ee006c7c41de56eafe80af905/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1077): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1201](https://github.com/Minecolonies/minecolonies/blob/945b86d195fe355ee006c7c41de56eafe80af905/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1201): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1203](https://github.com/Minecolonies/minecolonies/blob/945b86d195fe355ee006c7c41de56eafe80af905/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1203): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1205](https://github.com/Minecolonies/minecolonies/blob/945b86d195fe355ee006c7c41de56eafe80af905/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1205): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1398](https://github.com/Minecolonies/minecolonies/blob/945b86d195fe355ee006c7c41de56eafe80af905/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1398): Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**